packages/styling/src/interfaces/ISemanticColors.ts

```diff
@@ -1,3 +1,147 @@
+export interface ISemanticTextColors {
+  /* ANY ADDITIONS/REMOVALS HERE MUST ALSO BE MADE TO \packages\office-ui-fabric-react\src\common\_semanticSlots.scss */
+
+  //// Base slots
+
+  /**
+   * The default color for text.
+   */
+  bodyText: string;
+
+  /**
+   * Checked text color, e.g. selected menu item text.
+   */
+  bodyTextChecked: string;
+
+  /**
+   * De-emphasized text; e.g. metadata, captions, placeholder text.
+   */
+  bodySubtext: string;
+
+  /**
+   * Neutral colored links and links for action buttons.
+   */
+  actionLink: string;
+
+  /**
+   * Hover state for neutral colored links and links for action buttons.
+   */
+  actionLinkHovered: string;
+
+  /**
+   * The color of a link.
+   */
+  link: string;
+
+  /**
+   * The color of a hovered link. Also used when the link is active.
+   */
+  linkHovered: string;
+
+  /**
+   * The default color for disabled text on top of disabledBackground; e.g. text in a disabled text field, disabled button text.
+   */
+  disabledText: string;
+
+  /**
+   * The default color for disabled text on the default background (bodyBackground).
+   */
+  disabledBodyText: string;
+
+  /**
+   * Disabled de-emphasized text, for use on disabledBackground.
+   */
+  disabledSubtext: string;
+
+  //// Invariants - slots that rarely change color theme-to-theme because the color has meaning
+
+  /**
+   * The default color of error text, used on bodyBackground.
+   */
+  errorText: string;
+  /**
+   * The color of text on errorBackground, warningBackground, blockingBackground, or successBackground.
+   */
+  warningText: string;
+
+  /**
+   * The color of placeholder text.
+   */
+  inputPlaceholderText: string;
+
+  //// Buttons
+
+  /**
+   * Color of text in a standard button
+   */
+  buttonText: string;
```

Comment:
Button slots should be moved back into ISemanticColors, they shouldn't be available to use outside of a button context since they depend on the background color being a specific thing. (I am assuming the text component doesn't let you set backgrounds?) Same with warningText.